sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -952,63 +871,34 @@ public final FileResult close() throws Exception {
   }
 
   /**
-   * Result of a single bundle write. Contains the filename of the bundle.
+   * Result of a single bundle write. Contains the filename produced by the bundle, and if known
+   * the final output filename.
    */
-  public static final class FileResult {
-    private final String filename;
-    private final String destinationFilename;
+  public static final class FileResult implements Serializable {
+    private final ResourceId filename;
+    @Nullable private final ResourceId destinationFilename;
 
-    public FileResult(String filename, String destinationFilename) {
+    public FileResult(ResourceId filename, @Nullable ResourceId destinationFilename) {
       this.filename = filename;
       this.destinationFilename = destinationFilename;
     }
 
-    public String getFilename() {
+    public ResourceId getFilename() {
       return filename;
     }
 
-    public String getDestinationFilename() {
+    @Nullable public ResourceId getDestinationFilename() {
       return destinationFilename;
     }
 
-  }
-
-  /**
-   * A coder for FileResult objects.
-   */
-  public static final class FileResultCoder extends CustomCoder<FileResult> {
```

Comment:
Did you delete this because of ResourceId? This will break streaming update in a fundamental way. Compatibility checks will succeed (as the coder hasn't changed), but Java may fail to deserialize objects after update.